spec/requests/onboarding/telegram_spec.rb

```diff
@@ -173,78 +114,63 @@
+        let(:json_web_token) { create(:json_web_token, invalidated_jwt: jwt) }
+        before { json_web_token }
+        it { is_expected.to have_http_status(:unauthorized) }
+
+        context 'with a corresponding contributor' do
+          before { contributor }
+          describe 'who is not yet connected via Telegram' do
+            let(:contributor) do
+              create(:contributor, telegram_onboarding_token: 'TELEGRAM_ONBOARDING_TOKEN', telegram_id: nil, json_web_token: json_web_token)
+            end
+            it { is_expected.to have_http_status(:ok) }
+
+            describe 'but the jwt query parameter is invalid' do
+              let(:jwt) { 'INCORRECT_TOKEN' }
+              it { is_expected.to have_http_status(:unauthorized) }
+            end
+          end
 
-        it 'is unauthorized' do
-          passed_expiration_time = 31.minutes.from_now
-          Timecop.travel(passed_expiration_time) do
-            subject.call
-            expect(response).to be_unauthorized
+          describe 'who is already connected via Telegram' do
+            let(:contributor) { create(:contributor, telegram_onboarding_token: 'TELEGRAM_ONBOARDING_TOKEN', telegram_id: 4711) }
+            it { is_expected.to have_http_status(:unauthorized) }
           end
         end
       end
+    end
+  end
 
-      context 'without data processing consent' do
-        before { setup_telegram_id_cookie(contributor) }
-        let(:data_processing_consent) { false }
-        it 'displays validation errors' do
-          subject.call
-          parsed = Capybara::Node::Simple.new(response.body)
-          fields = parsed.all('.Field')
-          data_processing_consent_field = fields.find { |f| f.has_text? 'Allgemeine Nutzungsbedingungen' }
-          expect(data_processing_consent_field).to have_text('muss akzeptiert werden')
-        end
-      end
+  describe 'GET /onboarding/telegram/fallback' do
+    let(:jwt) { JsonWebToken.encode({ invite_code: 'ONBOARDING_TOKEN', action: 'onboarding' }) }
+    let(:params) { { telegram_onboarding_token: 'TELEGRAM_ONBOARDING_TOKEN', jwt: jwt } }
+    let(:json_web_token) { create(:json_web_token, invalidated_jwt: jwt) }
+    let(:contributor) do
+      create(:contributor, telegram_onboarding_token: 'TELEGRAM_ONBOARDING_TOKEN', telegram_id: nil, json_web_token: json_web_token)
     end
+    before { contributor }
+    let(:action) { -> { get onboarding_telegram_fallback_path(params) } }
+    subject { action.call && response }
 
-    context 'valid' do
-      before { setup_telegram_id_cookie(contributor) }
+    describe 'redirects' do
+      it 'are skipped' do
+        is_expected.to have_http_status(:ok)
+      end
 
-      it 'updates the contributor' do
-        subject.call
-        contributor = Contributor.find_by(telegram_id: 789)
-        expect(contributor).to have_attributes(first_name: 'Update', last_name: 'MyNames')
+      describe '(sanity check: / redirects to /link)' do
```

Comment:
I don’t really understand why this test is in this place, as a sanity check? Shouldn’t this either be a separate test for the `GET /onboarding/telegram` request? (If necessary at all, I think it’s covered by other requests, right?)

Reply:
This is a sanity check for the test setup. By changing `let(:action)` I make sure the test setup is alright and stays up-to-date even after we (maybe) refactor the routing behaviour.